ic-agent/src/agent/http_transport.rs

```diff
@@ -240,3 +250,30 @@ impl super::ReplicaV2Transport for ReqwestHttpReplicaV2Transport {
         Box::pin(run(self))
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::ReqwestHttpReplicaV2Transport;
+
+    #[test]
+    fn redirect() {
+        fn test(base: &str, result: &str) {
+            let t = ReqwestHttpReplicaV2Transport::create(base).unwrap();
+            assert_eq!(t.url.as_str(), result, "{}", base);
+        }
+
+        test("https://ic0.app", "https://ic0.app/api/v2/");
+        test("https://IC0.app", "https://ic0.app/api/v2/");
+        test("https://foo.ic0.app", "https://ic0.app/api/v2/");
```

Comment:
What would be the expected behavior?
test("https://fooic0.app", "https://???/api/v2/");

Reply:
Good catch. Should be unchanged. Fixed and tested in next revision.